underfs/s3/src/main/java/tachyon/underfs/s3/S3UnderFileSystemFactory.java

```diff
@@ -43,12 +44,16 @@ public UnderFileSystem create(String path, TachyonConf tachyonConf, Object unuse
 
     if (addAndCheckAWSCredentials(tachyonConf)) {
       TachyonURI uri = new TachyonURI(path);
-      return new S3UnderFileSystem(uri.getHost(), tachyonConf);
-    } else {
-      String err = "AWS Credentials not available, cannot create S3 Under File System.";
-      LOG.error(err);
-      throw Throwables.propagate(new IOException(err));
+      try {
+        return new S3UnderFileSystem(uri.getHost(), tachyonConf);
+      } catch (ServiceException se) {
+        // Do nothing, will fail after.
```

Comment:
if `ServiceException` is thrown, it means the credential is all right, but there is something wrong with creating the service. Therefore, I don't think you can rely on the "fail after", because it gives wrong error message.